### PR TITLE
Add diamond -> coal conversion

### PR DESCRIPTION
**Pull request by svr333**
## Description
Add one recipe to grindstone (and electric ore grinder)

## Proposed changes
- Add diamond to 24 coal conversion

## Related Issues (if applicable)
[Discord suggestion 774](https://discord.com/channels/565557184348422174/627048923122499584/822003610140409876)

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/GrindStone.java [hunk 1]
```diff
@@ -91,6 +91,9 @@ protected void registerDefaultRecipes(@Nonnull List<ItemStack> recipes) {
 
         recipes.add(SlimefunItems.ENDER_LUMP_3);
         recipes.add(new SlimefunItemStack(SlimefunItems.ENDER_LUMP_2, 4));
+
+        recipes.add(new ItemStack(Material.DIAMOND));
+        recipes.add(new ItemStack(Material.COAL, 24));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/GrindStone.java:96]: Why 24 though?

Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/GrindStone.java:96]: I felt like 16 was a bit too few and 32 a bit too much
What would you suggest?

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/GrindStone.java:96]: I did the maths, and it seems like 1 diamond (synthesized) corresponds to 8 Compressed Carbon which each correspond to 4 Carbon which each correspond to 8 coal.
That means 1 diamond is roughly worth 4 stacks of coal.
Making this a 1:1 mapping seems ridicilous, I admit.
But I think 32 is a bit more on the fair side here while still yielding a big loss of coal compared to the other way around.

Another idea on how to spice this up is to introduce a middle item.
How about you grind diamonds into e.g. 8 diamond-coal-hybrid item and then these can each be ground into 4 coal.
Or similar.

Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/GrindStone.java:96]: All right

Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/GrindStone.java:96]: I was looking for a good name and then I realized we already have Carbons, we can just grind a diamond into 4 Carbon, and then just add a way to convert 1 carbon back into 8 coal.
That seem good?

**Review by Semisol — CHANGES_REQUESTED** [review]
Looks good to me to be fair. I feel like the refund amount maybe might need an improvement but yeah.

But requesting changes because: The ore crusher can convert carbon to coal at a 1:1.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/GrindStone.java [hunk 2]
```diff
@@ -91,6 +91,12 @@ protected void registerDefaultRecipes(@Nonnull List<ItemStack> recipes) {
 
         recipes.add(SlimefunItems.ENDER_LUMP_3);
         recipes.add(new SlimefunItemStack(SlimefunItems.ENDER_LUMP_2, 4));
+
+        recipes.add(SlimefunItems.CARBON);
+        recipes.add(new ItemStack(Material.COAL, 8));
+
```
Comment (Semisol) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/GrindStone.java:97]: The ore crusher can already do this.
```suggestion
```

**Review by Semisol — APPROVED** [review]
Looks all good.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/GrindStone.java [hunk 3]
```diff
@@ -91,6 +91,8 @@ protected void registerDefaultRecipes(@Nonnull List<ItemStack> recipes) {
 
         recipes.add(SlimefunItems.ENDER_LUMP_3);
         recipes.add(new SlimefunItemStack(SlimefunItems.ENDER_LUMP_2, 4));
+        recipes.add(new ItemStack(Material.DIAMOND));
+        recipes.add(new SlimefunItemStack(SlimefunItems.CARBON, 4));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/GrindStone.java:95]:
```suggestion
        recipes.add(new SlimefunItemStack(SlimefunItems.ENDER_LUMP_2, 4));

        recipes.add(new ItemStack(Material.DIAMOND));
        recipes.add(new SlimefunItemStack(SlimefunItems.CARBON, 4));
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]
You didn't see anything :eyes:

**Review by Semisol — APPROVED** [review]

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**